src/main/webapp/app/modeling-assessment-editor/modeling-assessment-editor.component.ts

```diff
@@ -84,22 +81,20 @@ export class ModelingAssessmentEditorComponent implements OnInit, OnDestroy {
         });
         this.isAtLeastInstructor = this.accountService.hasAnyAuthorityDirect(['ROLE_ADMIN', 'ROLE_INSTRUCTOR']);
 
-        this.route.params.subscribe(params => {
-            const submissionId: String = params['submissionId'];
-            const exerciseId = Number(params['exerciseId']);
+        this.route.paramMap.subscribe(params => {
+            const submissionId: String | null = params.get('submissionId');
```

Comment:
In this case if `submissionId` is null we will call `loadSubmission`, which doesn't expect null. Is this intended?

Reply:
This is the same behavior we had before.
The null case is "handled" by the cast in Line 90:
```ts
this.loadSubmission(Number(submissionId));
```
Which converts `null` to `0`.